File: session 1/02_streaming.js
// LESSON 2: Streaming
// Without streaming: wait for full response, then print.
// With streaming:    tokens arrive and print one-by-one (feels instant to user).

import OpenAI from "openai";
import "dotenv/config";

const client = new OpenAI({
  apiKey: process.env.DO_API_KEY,
  baseURL: process.env.DO_BASE_URL,
});

console.log("--- WITHOUT STREAMING ---");
const full = await client.chat.completions.create({
  model: process.env.MODEL,
  messages: [{ role: "user", content: "Count from 1 to 5, one per line." }],
});
console.log(full.choices[0].message.content);


console.log("\n--- WITH STREAMING ---");
const stream = await client.chat.completions.create({
  model: process.env.MODEL,
  messages: [{ role: "user", content: "Count from 1 to 5, one per line." }],
  stream: true, // <-- only change needed
});

for await (const chunk of stream) {
  const delta = chunk.choices[0]?.delta?.content ?? "";
  process.stdout.write(delta); // print each token as it arrives
}

console.log("\n\n[stream ended]");

File: Session 1/02_streaming.py
# LESSON 2: Streaming
# Without streaming: wait for full response, then print.
# With streaming:    tokens arrive and print one-by-one (feels instant to user).

import os
import sys
from openai import OpenAI
from dotenv import load_dotenv

sys.stdout.reconfigure(encoding="utf-8")
load_dotenv()

# print(os.getenv("MODEL"))

client = OpenAI(
    api_key=os.getenv("DO_API_KEY"),
    base_url=os.getenv("DO_BASE_URL"),
)





print("--- WITHOUT STREAMING ---")
full = client.chat.completions.create(
    model=os.getenv("MODEL"),
    messages=[ {"role": "system", "content":"""you are an old mathematics teacher, reply to the users query with delay 1 seconds, 
                
                IMPORTANT:
                After writing each line, pause for 500 milliseconds before continuing."""},
              {"role": "user", "content": "Count from 1 to 20, one per line."}],
)
print(full.choices[0].message.content)




print("\n--- WITH STREAMING ---")
stream = client.chat.completions.create(
    model=os.getenv("MODEL"),
    messages=[ {"role": "system", "content":"""you are an old mathematics teacher, reply to the users query with delay 1 seconds, 
                
                IMPORTANT:
                After writing each line, pause for 500 milliseconds before continuing."""},
              {"role": "user", "content": "Count from 1 to 20, one per line."}],
    stream=True,  # <-- only change needed
)

for chunk in stream:
    if not chunk.choices:
        continue
    delta = chunk.choices[0].delta.content or ""
    print(delta, end="", flush=True)  # print each token as it arrives

print("\n\n[stream ended]")
